plugins/modules/foreman_smart_proxy.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_smart_proxy
+short_description: Manage Foreman Smart Proxy
+description:
+  - Manage Foreman Smart Proxy
+requirements:
+  - "nailgun >= 0.28.0"
```

Comment:
Have you considered, writing it with apypie as a backend? I think we are not going to accept any more nailgun based modules.
@evgeni can we state this somewhere visible?

Reply:
No idea where. We could add it as a PR template maybe?

Reply:
Better that now. But ideally, i'd like the contributor to see it before even starting to write the module.